connect/src/com/telenor/connect/headerenrichment/HeLogic.java

```diff
@@ -0,0 +1,210 @@
+package com.telenor.connect.headerenrichment;
+
+import android.annotation.TargetApi;
+import android.content.Context;
+import android.net.ConnectivityManager;
+import android.net.Network;
+import android.net.NetworkCapabilities;
+import android.net.NetworkInfo;
+import android.net.NetworkRequest;
+import android.os.Build;
+
+import com.telenor.connect.ConnectSdk;
+import com.telenor.connect.utils.ConnectUrlHelper;
+
+import java.util.Date;
+import java.util.Map;
+
+public class HeLogic {
+    private static final long HE_TOKEN_TIMEOUT_MILLISECONDS = 10_000;
```

Comment:
As an outsider, I'm curious as to why we use 10 000 here. However, not sure if it makes sense to add a comment. What do you think? It may make it easier to contribute to the project.

Reply:
Don't know... I guess it's implied that the number should be "long enough to succeed and short enough for users to wait it out if it should fail/time out". My best suggestion would be to reuse what you said: "Timeout is a best guess at what would be long enough to succeed and short enough for users to wait it out if it should fail/time out, and do a normal flow."

Reply:
Fixed: b27acf7